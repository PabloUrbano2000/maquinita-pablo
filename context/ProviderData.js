import React, { useState, createContext, useEffect } from 'react';

export const DATA = {
  information: {
    messageOne: '¡PRUEBA TU SUERTE HOY!',
    messageTwo: 'Si te salen los tres',
    messageThree: 'de Trome',
    messageFour: 'ganas el premiazo',
    messageFive: '¡AL INSTANTE!'
  },
  buttons: {
    main: '¡HAZ CLIC AQUÍ!',
    terms: 'TÉRMINOS Y CONDICIONES',
    more: 'CONOCE MÁS',
    participates: 'PARTICIPA'
  },
  knowMore: {
    messageOne: 'Tu puedes ser uno de los ganadores de',
    messageTwo: '4 premios de s/',
    messageThree: '200',
    messageFour: 'Preguntas frecuentes:',
    questions: [
      {
        question: '¿Cómo se si soy el ganador?',
        answer: `Si al girar la ruleta te aparecen las tres T de trome,
      <b>GANASTE!</b> Automáticamente te aparecerá el código
      ganador.`
      },
      {
        question: '¿Cuánto tiempo tengo que esperar por mi premio?',
        answer: `En un plazo de <b>60 días hábiles</b> nos estaremos
      contactando contigo para coordinar la entrega del premio.`
      },
      {
        question: '¿Si gané hoy día puedo ganar cualquier otro?',
        answer: `Si! En esta promoción puedes participar <b>todos los días.</b>`
      },
      {
        question: '¿Puedo compartir mi código ganador con algún familiar?',
        answer: `No, el código ganador es <b>personal</b> y está asociado a tu
        número de DNI`
      },
      {
        question: 'Perdí mi código, ¿cómo puedo recuperarlo?',
        answer: `<b>Escríbele a Fito al 977 944 966</b> y él podrá ayudarte a
        recuperar tú código.`
      }
    ]
  },
  game: {
    messageOne: 'Prueba tu suerte HOY y participa de',
    buttonJ: 'Jugar',
    buttonD: 'Detener'
  },
  informationDataUser: {
    messageOne: 'Ya casi terminas',
    messageTwo: 'Completa los datos y participa por el premiazo',
    ButtonOne: 'Clic Aquí',
    ButtonTwo: 'y completa tus datos'
  }
};

export const DataContext = createContext({
  systemData: '',
  setSystemData: () => {},
  auth: '',
  setAuth: () => {}
});

const ProviderData = ({ children }) => {
  const [systemData, setSystemData] = useState(DATA);
  const [auth, setAuth] = useState(null);

  useEffect(() => {
    window.tp.push([
      'init',
      function () {
        // esta función se ejecuta en cola, luego de que Piano se inicialice
        // console.log('hola', window.tp.pianoId.isUserValid());
        if (window.tpuser) {
          setAuth(window.tpuser);
        }
      }
    ]);
  }, []);

  const context = {
    systemData,
    setSystemData,
    auth,
    setAuth
  };

  return (
    <DataContext.Provider value={context}>{children}</DataContext.Provider>
  );
};

export default ProviderData;
