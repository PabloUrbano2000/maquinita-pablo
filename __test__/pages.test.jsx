import { render } from '@testing-library/react';
import SlotMachine from '../pages/slot-machine/index';
import ResultGame from '../pages/result-game/index';
import KnowMore from '../pages/know-more/index';
import Home from '../pages/index';
import { DATA } from '../context/ProviderData';

it('render page Home', () => {
  const { container } = render(<Home />);
  expect(container).toMatchSnapshot();
});

it('render page SlotMachine', () => {
  const { container } = render(<SlotMachine />);
  expect(container).toMatchSnapshot();
});

it('render page ResultGame', () => {
  const { container } = render(<ResultGame />);
  expect(container).toMatchSnapshot();
});

it('render page KnowMore', () => {
  const { container } = render(<KnowMore />);
  expect(container).toMatchSnapshot();
});
