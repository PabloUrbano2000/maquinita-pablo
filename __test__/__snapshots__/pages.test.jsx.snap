// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render page ResultGame 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1u9qabl-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1tfennd-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-1qb3ai4"
        >
          <div
            class="MuiBox-root css-1lgihvy"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%270%27%20height=%270%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="img-slot-t"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-8ffmfy-MuiPaper-root"
        >
          <div
            class="MuiBox-root css-18z1sft"
          >
            <div
              class="MuiBox-root css-0"
              style="transform: rotate(-3deg);"
            >
              <span
                class="css-1sah1m0"
                style="font-size: 2.3rem; display: block; margin-bottom: 5px; text-align: center;"
              >
                !FELICITACIONES!
              </span>
            </div>
            <span
              class="css-8rchdx"
              style="font-size: 1.3rem; text-align: center;"
            >
              Eres el afortunado ganador en llevarse el
               
              <span
                class="css-yzbrn7"
                style="font-size: 1.5rem; text-align: center;"
              >
                PREMIAZO DE TROME.
              </span>
            </span>
            <span
              class="css-yzbrn7"
              style="display: block; font-size: 1.5rem; text-align: center; color: rgb(255, 175, 80);"
            >
              NOMBRE Y APELLIDO
              <span
                class="css-yzbrn7"
                style="display: block; font-size: 1.5rem; text-align: center; color: rgb(255, 175, 80);"
              >
                DNI: 123123131
              </span>
            </span>
            <span
              class="css-yzbrn7"
              style="display: block; font-size: 1.3rem; text-align: center;"
            >
              Atento, porque estaremos enviando el código de validación a tu correo.
            </span>
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
              style="font-size: 1.3rem; border-radius: 80px;"
              tabindex="0"
              type="button"
            >
              Enviar código al correo
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
